transmart-api-server/grails-app/conf/application.yml

```diff
@@ -109,19 +109,10 @@ server:
 security:
     oauth2:
         client:
-            clientId: client_1
-#        resource:
-#            userInfoUri: http://localhost:8080/auth/realms/transmart/protocol/openid-connect/userinfo
-
-#keycloak:
-#    enable-basic-auth: false  # basic auth is default true, we dont want that
-#    realm: transmart   # our keycloak realm this ressource service should be based in
-#    auth-server-url: http://localhost:8080/auth  # the url to our sso server (keycloak)
-#    ssl-required: external  # no ssl is ok for demo  - do not use this in production
-#    resource: tm_client  # the realm-client which accessing clients should be authenticate with
-#    use-resource-role-mappings: true # use this if you manage your roles in client scope not in realm scope
-#    public-client: false # we do not write a web-app - so no login page and redirects are necessary
-#    bearer-only: true  # authentification via accesstoken is fine
-#    credentials:
-#        secret: 7034b121-fdf1-4188-9ba2-c66bee4b9867
+            clientId: <your-client-id>
```

Comment:
hm, is it sufficient to know only the clientId to make transmart functioning as the resource servers?
It does not have a URL to make a back channel call to the keycloak. Which is fine. But it does not seem to have means to get publick key of the keycloak instance.

Reply:
I am not sure, everything seems working only with this and using also the offlineToken, when doing a rest call to keycloak, but maybe there is a case, where we need also the secret.

Reply:
but maybe it is still using the keycloak.json config, I did not remove the KeycloakRestTemplate settings.